### PR TITLE
DSD-1224: updated color values

### PR DESCRIPTION
**Pull request by bigfishdesign13**
Fixes JIRA ticket [DSD-1224](https://jira.nypl.org/browse/DSD-1224)

## This PR does the following:

- Updates the hex value for `ui.gray.xx-dark`, `ui.gray.xxx-dark`, `ui.gray.xxxx-dark`, `dark.ui.bg.page`, `dark.ui.bg.hover`, `dark.ui.bg.active`, `dark.ui.disabled.secondary`, `dark.ui.error.primary`, `dark.ui.error.secondary`, `dark.ui.focus`, `dark.ui.link.primary`, `dark.ui.link.secondary`, `dark.ui.status.primary`, `dark.ui.status.secondary`, `dark.ui.success.primary`, `dark.ui.success.secondary`, `dark.ui.warning.primary` and `dark.ui.warning.secondary`.
- Updates the `Color Style Guide` to include new contrast details for the updated color values.

## How has this been tested?

- local Storybook

## Accessibility concerns or updates

- Color values have been vetted.

### Checklist:

- [x] I have updated the Storybook documentation accordingly.
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

**Review by EdwinGuzman — COMMENTED** [review]
Based on ticket DSD-1219, it seems that the "Background" colors needed to be updated for `dark.ui.bg.page`, `dark.ui.bg.hover`,  and `dark.ui.bg.active`. There's a slight mismatch in the `colors.ts` file. These values should also be included in the changelog.

**Comment by bigfishdesign13** [comment]
> Based on ticket DSD-1219, it seems that the "Background" colors needed to be updated for `dark.ui.bg.page`, `dark.ui.bg.hover`, and `dark.ui.bg.active`. There's a slight mismatch in the `colors.ts` file. These values should also be included in the changelog.

@EdwinGuzman,
`dark.ui.bg.page`, `dark.ui.bg.hover`, and `dark.ui.bg.active` are derived from the `gray` colors, which were updated, but yes, I should list the `page` colors in the CHANGELOG as well.  

What is the slight mismatch you are referring to?

**Comment by EdwinGuzman** [comment]
> What is the slight mismatch you are referring to?

DSD-1219:
```
dark.ui.bg.page: #191919
dark.ui.bg.hover: #252525
dark.ui.bg.active: #2E2E2E
```

DS:
```
const grayxxxxDark = "#191919";
const grayxxxDark = "#252525";
const grayxxDark = "#2E2E2E";
const grayxDark = "#424242";

// ...
bg: {
        page: grayxxxxDark, // this is fine
        default: grayxxxDark, // this is set to #252525 but is not defined in DSD-1219
        hover: grayxxDark, // this is set to #2e2e2e but should be #252525
        active: grayxDark, // this is set to #424242 but should be #2e2e2e defined in DSD-1219
      },
```

**Comment by bigfishdesign13** [comment]
> ```
> bg: {
>     page: grayxxxxDark, // this is fine
>     default: grayxxxDark, // this is set to #252525 but is not defined in DSD-1219
>     hover: grayxxDark, // this is set to #2e2e2e but should be #252525
>     active: grayxDark, // this is set to #424242 but should be #2e2e2e defined in DSD-1219
>  },
> ```

@EdwinGuzman
That is very much my mistake.  It was implemented properly on the code side only because I knew in my head what it was supposed to be.  I did not refer back to the ticket after I wrote it.  `hover` should have been `default` and `active` should have been `hover`.  To correct this, I just updated DSD-1219 to the following:

- dark.ui.bg.page: #191919 (same as ui.gray.xxxx-dark)
- dark.ui.bg.default: #252525 (same as ui.gray.xxx-dark)
- dark.ui.bg.hover: #2E2E2E (same as ui.gray.xx-dark)

**Review by EdwinGuzman — APPROVED** [review]

File: src/utils/utils.ts [hunk 1]
```diff
@@ -95,3 +95,26 @@ export const hexToRGB = (hex: string, alpha: number) => {
   const rgb = `${r}, ${g}, ${b}`;
   return alpha ? `rgba(${rgb},${alpha})` : `rgb(${rgb})`;
 };
+
+/** Calculate color luminance */
+export const colorLuminance = (r, g, b) => {
```
Comment (EdwinGuzman) [inline src/utils/utils.ts:100]: Forgot to ask, are these standard functions for color luminance and contrast ratio for the general web? Just wondering if there should be a note to say that these are functions used elsewhere in other non-nypl projects for luminance and color contrast, and a linked reference?

**Pull request merged**